### PR TITLE
Fix startup

### PR DESCRIPTION
**Pull request by sttts**
- register flags
- cleanup cmd code
- add test server and startup test
- add meta/v1 types to apiserver schemes

### CODE REVIEW

File: pkg/cmd/auth-apiserver/cmd.go [hunk 1]
```diff
@@ -3,14 +3,14 @@ package oauth_apiserver
 import (
 	"io"
 
+	"github.com/spf13/cobra"
```
Comment (p0lyn0mial) [inline pkg/cmd/auth-apiserver/cmd.go]: nit: add an empty space

File: pkg/cmd/auth-apiserver/cmd.go [hunk 2]
```diff
@@ -39,22 +39,44 @@ func NewOAuthAPIServerOptions(out io.Writer) *OAuthAPIServerOptions {
 	}
 }
 
+func (o OAuthAPIServerOptions) Validate(args []string) error {
```
Comment (p0lyn0mial) [inline pkg/cmd/auth-apiserver/cmd.go]: nit: should be a pointer receiver?

File: pkg/cmd/auth-apiserver/cmd.go (hunk 1 above)
Comment (sttts) [reply pkg/cmd/auth-apiserver/cmd.go]: How many import section do we want? 5? 6? :)

File: pkg/cmd/auth-apiserver/cmd.go (hunk 2 above)
Comment (sttts) [reply pkg/cmd/auth-apiserver/cmd.go]: No, shouldn't. It's must not have side effects.

File: pkg/cmd/auth-apiserver/cmd.go (hunk 1 above)
Comment (p0lyn0mial) [reply pkg/cmd/auth-apiserver/cmd.go]: we usually have:
```
std

other pkgs

kube

openshift
```

File: pkg/cmd/auth-apiserver/cmd.go (hunk 2 above)
Comment (p0lyn0mial) [reply pkg/cmd/auth-apiserver/cmd.go]: interesting, what can cause side effects?

**Pull request merged**